src/System.Runtime.Extensions/src/System/AppDomain.cs

```diff
@@ -229,7 +227,7 @@ public void SetThreadPrincipal(IPrincipal principal)
             lock (_forLock) {
                 // Check that principal has not been set previously.
                 if (_defaultPrincipal != null)
-                    throw new PolicyException(SR.AppDomain_Policy_PrincipalTwice);
+                    throw new SystemException(SR.AppDomain_Policy_PrincipalTwice);
```

Comment:
I assume this means we'll end up throwing a different exception from desktop in this case, but that's it's such a minor difference we're ok with it?